apps/projects/app/components/Panel/ReviewApplication/ReviewApplication.js

```diff
@@ -34,12 +36,32 @@ const ReviewApplication = ({ issue, requestIndex, readOnly }) => {
   } = useAragonApi()
   const { closePanel } = usePanelManagement()
   const theme = useTheme()
-
   const [ feedback, setFeedback ] = useState('')
-  const [ index, setIndex ] = useState(requestIndex)
 
-  const updateFeedback = e => setFeedback(e.target.value)
+  const canReview = issue.workStatus !== 'fulfilled' && !readOnly
+
+  const typesToShow = []
+  if (issue.requestsData.findIndex(request => !('review' in request)) !== -1)
+    typesToShow.push(canReview ? 'Available for review' : 'Unreviewed')
+  if (issue.requestsData.findIndex(request => 'review' in request) !== -1)
+    typesToShow.push('Reviewed')
+
+  let indexTypeSelected = 0
+  if ('review' in issue.requestsData[requestIndex])
+    indexTypeSelected = (typesToShow[0] === 'Reviewed') ? 0 : 1
+
+  const [ indexType, setIndexType ] = useState(indexTypeSelected)
+
+  const requests = issue.requestsData.filter(r => {
+    return (typesToShow[indexType] === 'Reviewed') ? 'review' in r : !('review' in r)
+  })
 
+  const newRequestIndex = requests.findIndex(
+    request => request.contributorAddr === issue.requestsData[requestIndex].contributorAddr
+  )
+  const [ index, setIndex ] = useState(newRequestIndex)
```

Comment:
Two things:

1. Let's rename these variables. `index` is generic to the point of meaninglessness.
2. We don't need to calculate `newRequestIndex` on every render, we only need to calculate it on first render, when we set `useState`. We can move the `findIndex` inline and avoid the clumsy name of `newRequestIndex`

Combining these, maybe we end up with:

```js
const [ selectedItem, setSelectedItem ] = useState(requests.findIndex(request =>
  request.contributorAddr === issue.requestsData[requestIndex].contributorAddr
))
```

But actually, something that would be **way** nicer, if it works, would be:

```js
const [ selectedItem, setSelectedItem ] = useState(requests.findIndex(request =>
  request === issue.requestsData[requestIndex]
))
```

I think this might work, given how object identity in JavaScript works. Please try it!

Reply:
It does seem to work indeed :)